commands/command-storage.ts

```diff
@@ -0,0 +1,71 @@
+export interface CommandStorageEntry<
+  TValue extends boolean | string = boolean | string,
+> {
+  readonly name: string;
+  readonly value: TValue;
+  readonly options?: any;
+}
+
+export class CommandStorage {
```

Comment:
very nitpicky but command storage name makes me think it somewhat holds the "command" reference while this isn't true (it represents the command context/options)

Reply:
good point

naming is hard, you know haha

I'll think of another name for it